packages/launch/src/focused-launch/summary/index.tsx

```diff
@@ -252,14 +274,13 @@ const Summary: React.FunctionComponent = () => {
 	);
 	const renderPlanStep = ( index: number ) => <PlanStep stepIndex={ index } key={ index } />;
 
-	// Steps that are not interactive (e.g. user has already selected domain/plan)
-	// Steps that are not interactive (e.g. user has already selected domain/plan)
-	const disabledSteps = hasPaidDomain ? [ renderDomainStep ] : [];
-
-	// Steps that require the user interaction
-	const activeSteps = hasPaidDomain
-		? [ renderSiteNameStep, renderPlanStep ]
-		: [ renderSiteNameStep, renderDomainStep, renderPlanStep ];
+	// Disabled steps are not interactive (e.g. user has already selected domain/plan)
+	// Active steps require user interaction
+	const disabledSteps: ( ( index: number ) => ReactNode )[] = [];
+	const activeSteps: ( ( index: number ) => ReactNode )[] = [];
+	shouldRenderNameStep && activeSteps.push( renderSiteNameStep );
+	( hasPaidDomain ? disabledSteps : activeSteps ).push( renderDomainStep );
+	( hasPaidPlan ? disabledSteps : activeSteps ).push( renderPlanStep );
```

Comment:
Aren't we constantly pushing items to these 2 Arrays with each update of `Summary` component?

Reply:
Yes, we are pushing to those arrays every render. But those arrays are just temporary variables (they are initialized on lines 279-280), and they offer a way to sort those steps correctly, while they're being split into two groups.

Do you see this implementation as problematic?

Reply:
Ok, now I see what are you trying to solve with this. Maybe the sorting objective should also be mentioned in the comment. What do you think of adding some unit tests for this bit? Not a blocker for this PR though.

Reply:
Added a comment, and made a point about unit-testing this piece of logic in #46943